### ISSUE TITLE
VMware: most modules give results warning

### ISSUE THREAD

**Issue by UnknownGnome** [ansible/ansible#62006]
##### SUMMARY

Most modules give the warning 
[WARNING]: Found internal 'results' key in module return, renamed to 'ansible_module_results'.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

vmware_host_service_manager

##### ANSIBLE VERSION

```paste below
ansible 2.8.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/awx/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jun 20 2019, 20:27:34) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

awx docker image

##### STEPS TO REPRODUCE

```yaml
    - name: enable SSH
      vmware_host_service_manager:
        hostname: "{{ vcenter }}"        
        username: "{{ lookup('env','VMWARE_USER') }}"
        password: "{{ lookup('env','VMWARE_PASSWORD') }}"
        esxi_hostname: "{{inventory_hostname}}"
        service_name: TSM-SSH
        state: start
```

##### ACTUAL RESULTS

```paste below
The module works fine, but it produces the warning
 [WARNING]: Found internal 'results' key in module return, renamed to
'ansible_module_results'.
```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @Akasurde @Tomorrow9 @goneri @lparkes @pgbidkar @warthog9
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@UnknownGnome, just so you are aware we have a dedicated Working Group for vmware.
You can find other people interested in this in `#ansible-vmware` on Freenode IRC
For more information about communities, meetings and agendas see https://github.com/ansible/community

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mariolenz** [comment]
Similar to issue #60275.

@Akasurde According to the documentation, you're the module authors. Can you please have a look? I don't understand what this `results` key is for, otherwise I'd try to do something about it. But I don't want to change things I don't understand.

**Comment by lparkes** [comment]
I expect that this is because the `results` key is now a standard return value used to bundle up the results of an ansible loop. See https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html#results 

I have to go AFK now, so I don't have time to look up anything more certain than Google.

The answer will be to simply rename results key manually to ansible_module_results and adjust any necessary documentation.

**Comment by mariolenz** [comment]
> The answer will be to simply rename results key manually to ansible_module_results and adjust any necessary documentation.

@Akasurde Can you please confirm that this is the fix? I'd try to implement it like this and open a PR if you do.

**Comment by Akasurde** [comment]
@mariolenz I think we should identify all modules in VMware space which uses this `results` key in return json and change appropriately. I am working on finding such modules and list down here. Once this is done we can decide the strategy to solve this issue. 

Let me know your views on this.

**Comment by UnknownGnome** [comment]
In addition to vmware_host_service_manager, I've seen the warning in vmware_host_ntp

**Comment by mariolenz** [comment]
@Akasurde 

> I think we should identify all modules in VMware space which uses this `results` key in return json

Issue #62083 (inspired by #59635)

> Once this is done we can decide the strategy to solve this issue.

Is there a default or best practice naming scheme for a variable that returns some kind of result from the module? There are some VMware modules that return a `result` without an `s`. Maybe we can just delete this one character.

**Comment by Akasurde** [comment]
Duplicate of https://github.com/ansible/ansible/issues/62083
